pkgs/applications/audio/zynaddsubfx/default.nix

```diff
@@ -1,30 +1,118 @@
-{ stdenv, fetchurl, alsaLib, cairo, cmake, libjack2, fftw, fltk13, lash,  libjpeg
-, libXpm, minixml, ntk, pkgconfig, zlib, liblo
+{ stdenv
+, lib
+, fetchFromGitHub
+, callPackage
+
+  # Required build tools
+, cmake
+, makeWrapper
+, pkgconfig
+
+  # Required dependencies
+, fftw
+, liblo
+, minixml
+, zlib
+
+  # Optional dependencies
+, alsaSupport ? true
+, alsaLib ? null
+, dssiSupport ? false
+, dssi ? null
+, ladspaH ? null
+, jackSupport ? true
+, libjack2 ? null
+, lashSupport ? false
+, lash ? null
+, ossSupport ? true
+, portaudioSupport ? true
+, portaudio ? null
+
+  # Optional GUI dependencies
+, guiModule ? "off"
+, cairo ? null
+, fltk13 ? null
+, libGL ? null
+, libjpeg ? null
+, libX11 ? null
+, libXpm ? null
+, ntk ? null
+
+  # Test dependencies
+, cxxtest
 }:
 
-stdenv.mkDerivation  rec {
+assert alsaSupport -> alsaLib != null;
+assert dssiSupport -> dssi != null && ladspaH != null;
+assert jackSupport -> libjack2 != null;
+assert lashSupport -> lash != null;
+assert portaudioSupport -> portaudio != null;
+
```

Comment:
```suggestion
```

We don't need this.

Reply:
I've seen other projects default optional dependencies to null and then assert them when their respective flag is enabled (for example curl: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/networking/curl/default.nix).

Shouldn't we be doing the same here?

Reply:
This is not required and makes no difference in the end.

Reply:
Ok, yeah I think the only case where this makes a difference is if one of the packages didn't exist, and it's feature wasn't enabled by default, but it would probably be better to get an error if it didn't. Thanks, I will submit another update.